packages/applications-service-api/src/services/representation.service.js

```diff
@@ -42,20 +46,57 @@ const createQueryFilters = (query) => {
 };
 
 const getRepresentationsForApplication = async (query) => {
+	return isBackOfficeCaseReference(query.caseReference)
+		? getRepresentationsForBO(query)
+		: getRepresentationsForNI(query);
+};
+
+const getRepresentationsForBO = async (query) => {
+	const options = {
+		caseReference: query.caseReference
+	};
+	const representations = await getRepresentationsByCaseReferenceBORepository(options);
+	const representationsData = await Promise.all(
+		representations.map(async (representation) => {
+			if (!representation.representedId) return;
+			const represented = await getServiceUserByIdBORepository(representation.representedId);
+			if (!represented) return;
+			const representative = await getServiceUserByIdBORepository(representation.representativeId);
+			return { representation, represented, representative };
+		})
+	);
+	const representationsDataWithRepresented = representationsData.filter((r) => r?.represented);
```

Comment:
Not sure if this filter is necessary, I think the block above enforces a represented to exist.

Reply:
The block above is a map function so it will return null but the null represented  element will still exist in the array. Then this line will remove that from the array so we don't return it in the response. 

More explanation: The reason I used the map above like this was to do promise.all and speed up concurrent requests.